### PR TITLE
NMT checkpoint averaging

### PR DESCRIPTION
**Pull request by michalivne**
This PR adds convenience scripts to checkpoint average any `.nemo` file (regardless of the class).
Python script is heavily based on code by @MaximumEntropy and @titu1994.

# Usage

The script `checkpoint_averaging.py` builds a `.nemo` file with average weights over multiple `.ckpt` files (assumes `.ckpt` files in same folder as `.nemo` file).
The input is a list of folders that contains a .nemo file (or a list of .nemo files).
The output of the script (per input folder / `.nemo` file) is a file named `*-averaged.nemo` where `*.nemo` is the original `.nemo` file.

# Example

Usage example for building `*-averaged.nemo` for a given `.nemo` file:

```
NeMo/scripts/checkpoint_averaging/checkpoint_averaging.py my_model.nemo
```

Usage example for building *-averaged.nemo files for all results in sub-directories under current path:

```
find . -name '*.nemo' | grep -v -- "-averaged.nemo" | xargs NeMo/scripts/checkpoint_averaging/checkpoint_averaging.py
```

### CODE REVIEW

**Comment by titu1994** [comment]
This already exists in a majority form here - https://github.com/NVIDIA/NeMo/blob/main/scripts/average_model_checkpoints.py

**Comment by MaximumEntropy** [comment]
Thanks @michalivne :) Maybe like Som suggested, we can move https://github.com/NVIDIA/NeMo/blob/main/scripts/average_model_checkpoints.py#L120:L167 to a separate function that gets called across both ways of using checkpoint averaging?

**Comment by michalivne** [comment]
@MaximumEntropy @titu1994 script is finished based on all comments. It is now very easy to. use, supposed to work with all nemo classes, and does not rely on a particular internal structure of `.nemo` files. Can you take a quick look?

File: scripts/checkpoint_averaging/checkpoint_averaging.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import torch
+
+from nemo.core import ModelPT
+from nemo.utils import logging, model_utils
+
+
+def main():
+    device = torch.device("cpu")
+
+    model_fname_list = sys.argv[1:]
```
Comment (titu1994) [inline scripts/checkpoint_averaging/checkpoint_averaging.py:40]: Why not just use argparse for future changes?

Comment (michalivne) [reply scripts/checkpoint_averaging/checkpoint_averaging.py:40]: I typically use it, but found it an overkill here.
I can use it here as well if you believe it to be useful.

Comment (michalivne) [reply scripts/checkpoint_averaging/checkpoint_averaging.py]: For the sake of uniformity I switched to argsparse.

**Review by titu1994 — DISMISSED** [review]
Looks great

**Review by titu1994 — DISMISSED** [review]

**Review by titu1994 — APPROVED** [review]

**Pull request merged**